testing/deployer/Makefile

```diff
@@ -0,0 +1,84 @@
+SHELL := /bin/bash
+
+.PHONY: noop
+noop:
+
+##@ Build
+
+.PHONY: tidy
+tidy: ## Run go mod tidy.
+	go mod tidy
+
+##@ Checks
+
+.PHONY: format
+format: ## Format the go files.
+	@for f in $$(find . -name '*.go' -print); do \
+		gofmt -s -w $$f ; \
+	done
+
+.PHONY: lint
+lint: ## Run the full linting rules.
+	golangci-lint run -v
+
+.PHONY: vet
+vet: ## Run go vet.
+	go vet ./...
+
+##@ Testing
+
+.PHONY: test
+test: ## Run all go tests.
+	go test ./...
+
+##@ Code Generation
+
+.PHONY: update-defaults ## Update default images versions for all relevant containers to the latest.
+update-defaults: update-consul update-envoy update-dataplane
+
+.PHONY: update-consul
+update-consul: ## Update the default images for Consul CE and Enterprise containers to the latest.
+	@docker pull hashicorp/consul:latest || true
+	@mkdir -p tmp
+	@docker run --rm hashicorp/consul:latest version -format=json | jq -r .Version > tmp/default_consul.val
+	@printf "// Copyright (c) HashiCorp, Inc.\n// SPDX-License-Identifier: BUSL-1.1\n\n" > topology/default_consul.go
+	@printf "package topology\n\nconst DefaultConsulImage = \"hashicorp/consul:$$(cat tmp/default_consul.val)\"\n" >> topology/default_consul.go
+	@printf "const DefaultConsulEnterpriseImage = \"hashicorp/consul-enterprise:$$(cat tmp/default_consul.val)-ent\"\n" >> topology/default_consul.go
+	@rm -rf tmp
+
+.PHONY: update-envoy
+update-envoy: ## Update the default image for the Envoy container to the latest.
+	@docker rm -f consul-envoy-check &>/dev/null || true
+	@docker pull hashicorp/consul:latest || true
+	@docker run -d --name consul-envoy-check hashicorp/consul:latest
+	@mkdir -p tmp
+	@docker exec consul-envoy-check sh -c 'wget -q localhost:8500/v1/agent/self -O -' | jq -r '.xDS.SupportedProxies.envoy[0]' > tmp/default_envoy.val
+	@docker rm -f consul-envoy-check &>/dev/null || true
+	@printf "// Copyright (c) HashiCorp, Inc.\n// SPDX-License-Identifier: BUSL-1.1\n\n" > topology/default_envoy.go
+	@printf "package topology\n\nconst DefaultEnvoyImage = \"envoyproxy/envoy:v$$(cat tmp/default_envoy.val)\"\n" >> topology/default_envoy.go
+	@rm -rf tmp
+
+.PHONY: update-dataplane
+update-dataplane: ## Update the default image for the Consul Dataplane container to the latest.
+	@docker pull hashicorp/consul-dataplane:latest || true
+	@mkdir -p tmp
+	@docker run --rm hashicorp/consul-dataplane:latest --version | head -n 1 | awk '{print $$3}' | cut -d'v' -f2- > tmp/default_dataplane.val
```

Comment:
Instead of running and string munging, can we not get this from docker tags or metadata?

Reply:
For consul ce/ent and dataplane the image labels work. For the envoy one we do still need to fire up that version of Consul to interrogate it for it's maximum supported envoy version.

Reply:
gross